### PR TITLE
Chapter 1 solutions

### PR DESCRIPTION
**Pull request by g-belmonte**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -477,8 +490,8 @@ Implement a function that returns the last digit of a given number.
   results. Or you can try to guess the function name, search for it and check
   whether it works for you!
 -}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = abs n `rem` 10
```
Comment (vrom911) [inline src/Chapter1.hs:494]: Nice work on using infix form of calling functions! We don't explain this syntax in our course to make it short. But well done :+1:

File: src/Chapter1.hs [hunk 2]
```diff
@@ -556,7 +572,9 @@ True
 >>> isVowel 'x'
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+isVowel :: Char -> Bool
+isVowel c =
+  c `elem` ['a', 'e', 'i', 'o', 'u']
```
Comment (vrom911) [inline src/Chapter1.hs]: Absolutely correct!

One note in here, that sometimes, `elem` could be slower than the explicit pattern matching. I remember there were some benchmarks on one particular case, that showed how moving to pattern matching on each case separately drastically decrease time 🐎

File: src/Chapter1.hs [hunk 3]
```diff
@@ -619,8 +637,12 @@ Implement a function that returns the sum of the last two digits of a number.
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n
+  | num < 10 = num
+  | otherwise = num `rem` 10 + (num `div` 10) `rem` 10
```
Comment (vrom911) [inline src/Chapter1.hs]: Yeap, great!
Note, that `mod x 10` (or rem that you use here) is the same as `lastDigit` function from the previous task 🙂

**Review by vrom911 — APPROVED** [review]
Great and thoughtful solutions 👍🏼 Congrats with finishing Chapter One 🥳 

Feel free to merge this PR as it is awesome 👍🏼

File: src/Chapter1.hs (hunk 2 above)
Comment (g-belmonte) [reply src/Chapter1.hs]: Спаcибо огромное за review! :grin: 
I have been tinkering with Haskell for a while, but I still don't know how to get the best performance out of it.
I'll change this function to a pattern matching one :+1:

File: src/Chapter1.hs (hunk 3 above)
Comment (g-belmonte) [reply src/Chapter1.hs]: Oh! Nice that you brought this up! Using lastDigit here will make the solution more elegant :smile:

**Pull request merged**